### PR TITLE
Provide Endpoint for role assignment to retrieve specific permissions

### PR DESCRIPTION
**Pull request by Abdu-moustafa**
This PR :
Introduces a new API endpoint `GET` `@role-assignment-report` that provides necessary information for generating role assignment reports in the new UI. It does not impact the existing `@role-assignments `and` @role-assignment-reports `endpoints, which will continue to function as they currently do.

For [TI-1543](https://4teamwork.atlassian.net/browse/TI-1543)

## Checklist

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

_Only applicable should be left and checked._

- Upgrade-Steps:
  - [ ] SQL Operations do not use imported model (see [docs](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/994344994/Upgrade-Steps))
  - [ ] Make it deferrable if possible
  - [ ] Execute as much as possible conditionally
  - DB-Schema migration
    - [ ] All changes on a model (columns, etc) are included in a DB-schema migration.
    - [ ] Constraint names are shorter than 30 characters (`Oracle`)
- API change:
  - [x] Documentation is updated
  - [x] API Changelog entry (see [guide](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/451248812/API+Changelog+Guidelines))
  - If breaking:
    - [ ] api-change label added
    - [ ] #delivery channel notified about breaking change
    - [ ] Scrum master is informed
- Bug fixed:
  - [ ] Resolved any Sentry issues caused by this bug
- New functionality:
  - [ ] for `document` also works for `mail`
  - [ ] for `task` also works for `forwarding`
- Further improvements needed:
  - [ ] Create follow-up stories and link them in the PR and Jira issue
- [ ] Change could impact client installations, client policies need to be adapted
- New translations
  - [ ] All msg-strings are unicode
- Change in schema definition:
  - [ ] If `missing_value` is specified, then `default` has to be set to the same value

[TI-1543]: https://4teamwork.atlassian.net/browse/TI-1543?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: opengever/api/role_assignment_reports.py [hunk 1]
```diff
@@ -151,3 +153,58 @@ def reply(self):
...
+
+
+class RoleAssignmentReportGet(Service):
+
+    def reply(self):
+        principal_id = self.request.form.get("principal_id")
+        include_membership = self.request.form.get("include_membership", False)
+        root = self.request.form.get("root")
+        b_start = self.request.form.get("b_start", 0)
+        b_size = self.request.form.get("b_size", 25)
```
Comment (elioschmutz) [inline opengever/api/role_assignment_reports.py:165]: Use `safe_int` for these variables. The query-string usually contains a string.

`from opengever.base.utils import safe_int`

File: opengever/api/role_assignment_reports.py [hunk 2]
```diff
@@ -151,3 +153,58 @@ def reply(self):
         storage.delete(report_id)
         self.request.response.setStatus(204)
         return super(RoleAssignmentReportsDelete, self).reply()
+
+
+class RoleAssignmentReportGet(Service):
+
+    def reply(self):
+        principal_id = self.request.form.get("principal_id")
+        include_membership = self.request.form.get("include_membership", False)
```
Comment (elioschmutz) [inline opengever/api/role_assignment_reports.py:162]:
```suggestion
        include_memberships = self.request.form.get("include_memberships", False)
```

File: opengever/api/role_assignment_reports.py [hunk 3]
```diff
@@ -151,3 +153,58 @@ def reply(self):
...
+        root = self.request.form.get("root")
+        b_start = self.request.form.get("b_start", 0)
+        b_size = self.request.form.get("b_size", 25)
+
+        response = {}
+
+        report = RoleAssignmentReporter()(
+            principal_id,
+            include_membership,
```
Comment (elioschmutz) [inline opengever/api/role_assignment_reports.py:171]:
```suggestion
            include_memberships,
```

File: docs/public/dev-manual/api/role_assignment_reports.rst [hunk 4]
```diff
@@ -176,6 +176,119 @@ Mittels DELETE-Requests kann ein Bericht gelöscht werden.
 
       HTTP/1.1 204 No content
 
+Berechtigungsreport-Abfrage
+-----------------------------
```
[the remark this answers is not in the document]
Comment (Abdu-moustafa) [reply docs/public/dev-manual/api/role_assignment_reports.rst:180]: ✔️

File: opengever/api/role_assignment_reports.py (hunk 1 above)
Comment (Abdu-moustafa) [reply opengever/api/role_assignment_reports.py:165]: ✔️

File: opengever/api/role_assignment_reports.py (hunk 3 above)
Comment (Abdu-moustafa) [reply opengever/api/role_assignment_reports.py:171]: ✔️

File: opengever/api/role_assignment_reports.py (hunk 2 above)
Comment (Abdu-moustafa) [reply opengever/api/role_assignment_reports.py:162]: ✔️

**Comment by elioschmutz** [comment]
@Abdu-moustafa you can rebase to the newest master. Everything is merged. And check your commit message. It seems to be wrong.

**Review by elioschmutz — APPROVED** [review]
👍 

rebase and rename your commit message. Use `git fetch; git rebase -i origin/master` for rebase and remove the commits which do not belong to your pr. This reduces merge conflicts. The commits are coming from force-pushes of the base-branches. Becuase the commit hash changes, git thinks, it's a new commit and tries to reapply it which will fail.

**Pull request merged**